BPH HLT DQM for 93x

anigamova opened the pull request.

Commit:
BPHHLT for 93x

cmsbuild commented:
A new Pull Request was created by  @anigamova  for master.

It involves the following packages:

DQMOffline/Trigger


@vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented:
tracked by #19142

cmsbuild commented:
+1
Tested at: 6a00bd0893045d05ee9e8f352cc968de2406ca14
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/21725/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3b3a9c06012a7e8edae545f64feb85061d66fab6
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/21725/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/21725/git-merge-result

cmsbuild commented:
Comparison job queued.

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/21725/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2354627
 - DQMHistoTests: Total failures: 14990
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2339471
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.h:
```diff
@@ -80,26 +96,94 @@ class BPHMonitor : public DQMEDAnalyzer
+  int trOrMu_;
+  int Jpsi_;
+  int Upsilon_;
+  int nofset_;
```
Hi @anigamova - thanks for the cleaned up PR - its easier to make comments on now.

Could nofset_ be an enum instead ? This would let its values describe what they actually mean.

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.h:
```diff
@@ -80,26 +96,94 @@ class BPHMonitor : public DQMEDAnalyzer
+
+
+///
+METME  phPhi_   ;
```
please make the indentation consistent.

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -111,30 +197,160 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+}
+
+/////
+//  DiMuPhi_.numerator   = nullptr;
```
please remove the comments - these look like obsolete lines

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.h:
```diff
@@ -80,26 +96,94 @@ class BPHMonitor : public DQMEDAnalyzer
+  StringCutObjectSelector<reco::Muon,true>        muoSelection_tag;
+  StringCutObjectSelector<reco::Muon,true>        muoSelection_probe;
   int nmuons_;
+  int tnp_;
```
it looks like this should be a bool?

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+    std::vector<reco::Muon> tagMuons;
+    std::vector<reco::Muon> probeMuons;
+    for ( auto const & m : *muoHandle ) {//applying tag selection 
+      if(!L3_){//matching does not work for L3 tnp monitoring yet 
```
it looks like this check can be moved outside of the for loop

davidlange6 replied:
if you nest the subsequent for loop into this one you can avoid making a copy of all your tag muons (eg, no need for tagMuons)

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+    if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+    iEvent.getByToken( hltInputTag_, handleTriggerEvent);
+    if (handleTriggerEvent->sizeFilters()== 0)return;
+    std::string hltpath = hltpaths_den[0]; 
```
this can be a const &

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+          if (!DMSelection_ref(m1.p4() + m.p4()))continue;
+          if (m.charge()*m1.charge()>0 )continue;
+        }
+        edm::ESHandle<MagneticField> bFieldHandle;
```
please move this such that it happens once per event

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+        reco::TransientTrack mu2TT(m1.track(), &(*bFieldHandle));
+        j_tks.push_back(mu1TT);
+        j_tks.push_back(mu2TT);
+        if (j_tks.size()!=2) continue;
```
this check is not needed (its by definition true)

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+        KalmanVertexFitter jkvf;
+        TransientVertex jtv = jkvf.vertex(j_tks);
+        if (!jtv.isValid()) continue;
+        reco::Vertex jpsivertex = jtv;
```
is  this copy needed?

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+          if(!matchToTrigger(hltpath,t, handleTriggerEvent)) continue;
+              reco::Track itrk1       = t ;                                                
+              
+              if((reco::deltaR(t.eta(), t.phi(),m1.eta(),m1.phi()) <= 0.001))continue;//checking overlaping
```
these can be reco::deltaR(t,m1) <= 0.001..

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+              t_tks.push_back(mu1TT);
+              t_tks.push_back(mu2TT);
+              t_tks.push_back(trTT);
+              if (t_tks.size()!=3) continue;
```
this is no needed.

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+              TransientVertex tv  = kvf.vertex(t_tks);
+              reco::Vertex vertex = tv;
+              if (!tv.isValid()) continue;
+              if( pB.mass()> maxmassJpsiTk || pB.mass()< minmassJpsiTk)continue;
```
it looks like this should be moved up to line 616

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+          for (auto const & t1 : *trHandle) {
+            if(!trSelection_ref(t1))continue;
+            if(!matchToTrigger(hltpath,t1, handleTriggerEvent)) continue;
+            reco::Track itrk1       = t ;                                                
```
these can be const &?

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+            reco::Track itrk2       = t1 ;                                                
+            if((reco::deltaR(t.eta(), t.phi(),m1.eta(),m1.phi()) <= 0.001))continue;//checking overlaping
+            if((reco::deltaR(t.eta(), t.phi(),t1.eta(),t1.phi()) <= 0.001))continue;//checking overlaping
+            if((reco::deltaR(t.eta(), t.phi(),m.eta(),m.phi()) <= 0.001)) continue;
```
similar comments about deltaR as above -

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+            TransientVertex tv  = kvf.vertex(t_tks);
+            reco::Vertex vertex = tv;
+            if (!tv.isValid()) continue;
+            if( pB.mass()> maxmassJpsiTk || pB.mass()< minmassJpsiTk)continue;
```
this can be moved to line 731?

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+  }
+
+   if (nofset_ == 7){//photons
+   std::string hltpath = hltpaths_den[0];
```
should be a const &

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+      reco::TransientTrack mu2TT(m1.track(), &(*bFieldHandle));
+      j_tks.push_back(mu1TT);
+      j_tks.push_back(mu2TT);
+      if (j_tks.size()!=2) continue;
```
not needed

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+      if(!muoSelection_ref(m1))continue;   
+      if(!matchToTrigger(hltpath,m1, handleTriggerEvent)) continue;
+      if (!DMSelection_ref(m1.p4() + m.p4()))continue;
+      edm::ESHandle<MagneticField> bFieldHandle;
```
move outside of the loops

davidlange6 commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,55 +362,787 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+          t_tks.push_back(mu1TT);
+          t_tks.push_back(mu2TT);
+          t_tks.push_back(trTT);
+          if (t_tks.size()!=3) continue;
```
similar comments in this part of the switch as elsewhere

davidlange6 commented:
-1
pending answers to comments.

Commit:
small modifications

Commit:
changes from Thomas

cmsbuild commented:
Pull request #19888 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

anigamova committed:
fixing a bug

cmsbuild commented:
Pull request #19888 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

anigamova commented:
Hi @davidlange6 !
Can you please take a look at the code?

fwyzard commented:
Please fix the conflicts.

anigamova committed:
Merge branch 'master' into from-CMSSW_9_3_X_2017-07-24-1100

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2ef559cf62912ca6b547ebd304399fed7f70d4e4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22120/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/2ef559cf62912ca6b547ebd304399fed7f70d4e4
https://github.com/cms-sw/cmssw/commit/909ea2f608171533a08bd310a0376fdbfe8d3137
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22120/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22120/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22120/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2642209
 - DQMHistoTests: Total failures: 228
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2641800
 - DQMHistoTests: Total skipped: 181
 - DQMHistoTests: Total Missing objects: 0
 - Checked 102 log files, 14 edm output root files, 25 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

mtosi commented:
can we merge this PR ?

Commit:
changes from the latest PR and modifications from Alessio

anigamova committed:
Merge branch 'from-CMSSW_9_3_X_2017-07-24-1100' into BPh_DQM_from_93X

anigamova committed:
Merge pull request #12 from anigamova/BPh_DQM_from_93X

adding modifications from Alessio

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/129

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/129/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/129/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

anigamova committed:
changing the type of two modules

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/135

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/135/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/135/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

tmadlener committed:
Merged BPh_DQM_from_93X from repository cms-bph-trigger

tmadlener committed:
Fix various small issues. Fix binnings.

tmadlener committed:
Add possibility to have a binning with user defined bin-edges.

- Use edges = cms.vdouble(...) to define this as single parameter in the ParameterSet
  defining the binning to have the values in the list taken as bin edges.
- Old way still works, but only one of the two ways can be defined at the same time
  otherwise there will be an exception during construction of the module.

tmadlener committed:
Fix some minor basic selection

anigamova committed:
fixing a bug

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/151

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/151/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/151/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

mtosi commented:
@cmsbuild, please test

> On 16 Aug 2017, at 08:28, cmsbuild <notifications@github.com> wrote:
> 
> -code-checks
> 
> Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/151
> 
> Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/151/git-diff.patch
> e.g. curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/151/git-diff.patch | patch -p1
> 
> In future, you can run scram build code-checks to apply code checks
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ac4e34d959c84eae6333a6b19159ff0cca9130a8
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22334/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/512d77ca964172c069f3cbe34930afa802c65b1d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22334/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22334/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22334/summary.html

Comparison Summary:
 - You potentially added 87051 lines to the logs
 - Reco comparison results: 54 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718134
 - DQMHistoTests: Total failures: 303
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2717642
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

aboletti committed:
Indentation fixed

aboletti committed:
Typo fixes in path names

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

anigamova committed:
Merge branch 'from-CMSSW_9_3_X_2017-07-24-1100' into BPH_DQM1_from930pre3

anigamova committed:
Merge pull request #13 from cms-bph-trigger/BPH_DQM1_from930pre3

Changes from Thomas and Alessio

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/266

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/266/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/266/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

anigamova commented:
can we please test it again?

mtosi commented:
is the BPH group done w/ this PR ?
or are you going to update it again ?
can you please summarize the list of triggers monitored by this PR ?

thanks

On Wed, Aug 23, 2017 at 4:03 PM, anigamova <notifications@github.com> wrote:

> can we please test it again?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19888#issuecomment-324342718>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58xYyWIUeYgSoBHtNvahwSq6NrcSjks5sbDFIgaJpZM4OhfpH>
> .
>

anigamova commented:
Yes, I think we are done with this PR

mtosi commented:
@cmsbuild, please test

On Fri, Aug 25, 2017 at 10:33 AM, anigamova <notifications@github.com>
wrote:

> Yes, I think we are done with this PR
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19888#issuecomment-324857334>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58y-tpMLDovz-OEPgnrENU8AIR_-eks5sboa-gaJpZM4OhfpH>
> .
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: d61acb747c00a27355aa32d758574c17d2ee679f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22490/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5d9892889f68e5ca315afc3b31869460844e9db2
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22490/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22490/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22490/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2652974
 - DQMHistoTests: Total failures: 265
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2652520
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

mtosi commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,60 +359,785 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+	      if((reco::deltaR(t,m) <= 0.001)) continue;
+	      if (! itrk1.quality(reco::TrackBase::highPurity))     continue;
+	      reco::Particle::LorentzVector pB, p2, p3;
+	      double trackMass2 = 0.493677 *0.493677;
```
please, avoid to have such kind of hard coded magic numbers

mtosi commented on DQMOffline/Trigger/plugins/BPHMonitor.cc:
```diff
@@ -146,60 +359,785 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+          cApp.calculate(mu1TS.theState(), mu2TS.theState());
+        }
+        double DiMuMass = (m1.p4()+m.p4()).M();
+        switch(enum_){//enum_ = 1...9, represents different sets of variables for different paths, we want to have different hists for different paths
```
don't you have other ways for handling this ?

mtosi replied:
are you totally sure all MEs are available ?

mtosi commented:
this PR makes the job crashing !!!!!
apparently, it is due to the module `DimuonX_HLT_OS_Vtx`

@anigamova @aboletti please fix this PR !
in addition, I think this code needs to be reviewed/rewritten
please, try to converge by next week

anigamova committed:
deleting crashing module

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/561

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/561/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/561/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly (this will soon be required for PRs to be considered)

mtosi commented:
@cmsbuild, please test

On Thu, Sep 7, 2017 at 5:32 PM, cmsbuild <notifications@github.com> wrote:

> -code-checks
>
> Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/561
>
> Code check has found code style and quality issues which could be resolved
> by applying a patch in https://cmssdt.cern.ch/SDT/
> code-checks/PR-19888/561/git-diff.patch
> e.g. curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/561/git-
> diff.patch | patch -p1
>
> You can run scram build code-checks to apply code checks directly (this
> will soon be required for PRs to be considered)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19888#issuecomment-327836421>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt585qVxMMF1KiCqN7LzTsERxLaaUpZks5sgAyFgaJpZM4OhfpH>
> .
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #19888 was updated. @ghellwig, @perrozzi, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @smuzaffar, @Dr15Jones, @kkousour, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @efeyazgan, @vazzolini, @rekovic, @kmaeshima, @arunhep, @emeschi, @thuer, @dmitrijus, @alja, @franzoni, @lpernie can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/565

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-19888/565/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-19888/565/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly (this will soon be required for PRs to be considered)

kpedro88 commented:
-1
see http://cms-sw.github.io/tutorial-resolve-conflicts.html#recovering-from-a-bad-rebase-or-merge to undo what you just did and then follow the rest of that page to update the PR correctly

Commit:
Merge branch 'from-CMSSW_9_3_X_2017-07-24-1100' of https://github.com/anigamova/cmssw into from-CMSSW_9_3_0_pre4

Commit:
making numbers configurable

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/568

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22788/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2641480
 - DQMHistoTests: Total failures: 217
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2641074
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

mtosi commented:
do we have any news on the needed fix ?

On Mon, Sep 11, 2017 at 2:04 PM, cmsbuild <notifications@github.com> wrote:

> This pull request is fully signed and it will be integrated in one of the
> next master IBs (tests are also fine). This pull request will now be
> reviewed by the release team before it's merged. @davidlange6
> <https://github.com/davidlange6>, @slava77 <https://github.com/slava77>,
> @smuzaffar <https://github.com/smuzaffar> (and backports should be raised
> in the release meeting by the corresponding L2)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19888#issuecomment-328508550>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt581GkDuMmOtIYg7dXAGE1B0UCeAZHks5shSHOgaJpZM4OhfpH>
> .
>

Commit:
prescale info added

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/651

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

mtosi commented:
@anigamova please let me know when you are done

Commit:
fixed segmentation violation for DimuonX_HLT_OS_Vtx, fixed a typo

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/652

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

mtosi commented:
@cmsbuild, please test

On Tue, Sep 12, 2017 at 6:50 PM, cmsbuild <notifications@github.com> wrote:

> +code-checks
>
> Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/652
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19888#issuecomment-328914459>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt584cImW80eXad3BoDluLZcajnSpn-ks5shrZcgaJpZM4OhfpH>
> .
>

cmsbuild commented:
The tests are being triggered in jenkins.

anigamova commented:
Hi @mtosi Alessio has spotted another problem giving segmentation violation for DimuonX_HLT_OS_Vtx module, so I have to remove it again.

anigamova committed:
removing DimuonX_HLT_OS_Vtx from the sequence

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19888/669

cmsbuild commented:
Pull request #19888 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 91632f7dc1f2be34de198aece8d84456c6e74027
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22933/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/8dd9a2c51f009a80e5374a9b450f6fa76f0f5189
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22933/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22933/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19888/22933/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2645564
 - DQMHistoTests: Total failures: 225
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2645150
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

dmitrijus commented:
+1
Please provide pull request description!

davidlange6 commented:
+1
